report_xml/hooks.py

```diff
@@ -0,0 +1,86 @@
+# License AGPL-3.0 or later (https://www.gnuorg/licenses/agpl.html).
+
+import os
+
+from odoo import SUPERUSER_ID, api
+
+
+# To create Environment inside a hook
+# ```
+# with api.Environment.manage():
+#     env = api.Environment(cr, SUPERUSER_ID, {})
+# ```
+
+
+def post_load():
+    """
+    Loaded before any model or data has been initialized.
+    This is ok as the post-load hook is for server-wide
+    (instead of registry-specific) functionalities.
+    This is very useful to create monkey patches for odoo.
+    Note: You do not have access to database cursor here.
+    """
+    pass
+
+
+def pre_init_hook(cr):
+    """
+    Loaded before installing this module, but after all previous modules
+    are installed.
+    None of this module's DB modifications will be available yet.
+
+    If you plan to raise an exception to abort install, put all code inside a
+    ``with cr.savepoint():`` block to avoid broken databases.
+
+    :param odoo.sql_db.Cursor cr:
+        Database cursor.
+    """
+    pass
+
+
+def post_init_hook(cr, registry):
+    """
+    Loaded after installing this module, and before the next module starts
+    installing.
+    This module's DB modifications will be available.
+
+    :param odoo.sql_db.Cursor cr:
+        Database cursor.
+    :param odoo.modules.registry.RegistryManager registry:
+        a mapping between model names and model classes.
+    """
+    with api.Environment.manage():
+        env = api.Environment(cr, SUPERUSER_ID, {})
+        demo_report = env["ir.actions.report"].search(
+            [("report_name", "=", "report_xml.demo_report_xml_view")],
+            limit=1,
+        )
+        if demo_report:
+            xsd_file_path = os.path.join(
+                os.path.dirname(__file__),
+                "demo/demo_report.xsd",
+            )
+            with open(xsd_file_path, "r") as xsd:
+                attach_vals = {
+                    "name": "Demo Report.xsd",
+                    "datas": xsd.read(),
+                    "res_model": "ir.actions.report",
+                    "res_id": demo_report.id,
+                    "res_field": "xsd_schema",
+                    "type": "binary",
+                }
+                env["ir.attachment"].create(attach_vals)
```

Comment:
Why do you need this?

Reply:
1. I left comment to previous similar question.
2. Docstring to function was updated